config/configgrpc/configgrpc.go

```diff
@@ -52,23 +69,54 @@ type GRPCClientSettings struct {
 
 	// The keepalive parameters for client gRPC. See grpc.WithKeepaliveParams
 	// (https://godoc.org/google.golang.org/grpc#WithKeepaliveParams).
-	KeepaliveParameters *KeepaliveConfig `mapstructure:"keepalive"`
+	Keepalive *KeepaliveClientConfig `mapstructure:"keepalive"`
 
 	// WaitForReady parameter configures client to wait for ready state before sending data.
 	// (https://github.com/grpc/grpc/blob/master/doc/wait-for-ready.md)
 	WaitForReady bool `mapstructure:"wait_for_ready"`
 }
 
-// KeepaliveConfig exposes the keepalive.ClientParameters to be used by the exporter.
-// Refer to the original data-structure for the meaning of each parameter.
-type KeepaliveConfig struct {
-	Time                time.Duration `mapstructure:"time,omitempty"`
-	Timeout             time.Duration `mapstructure:"timeout,omitempty"`
+type KeepaliveServerConfig struct {
+	ServerParameters  *KeepaliveServerParameters  `mapstructure:"server_parameters,omitempty"`
+	EnforcementPolicy *KeepaliveEnforcementPolicy `mapstructure:"enforcement_policy,omitempty"`
+}
+
+// KeepaliveServerParameters allow configuration of the keepalive.ServerParameters.
+// The same default values as keepalive.ServerParameters are applicable and get applied by the server.
+// See https://godoc.org/google.golang.org/grpc/keepalive#ServerParameters for details.
+type KeepaliveServerParameters struct {
+	MaxConnectionIdle     time.Duration `mapstructure:"max_connection_idle,omitempty"`
+	MaxConnectionAge      time.Duration `mapstructure:"max_connection_age,omitempty"`
+	MaxConnectionAgeGrace time.Duration `mapstructure:"max_connection_age_grace,omitempty"`
+	Time                  time.Duration `mapstructure:"time,omitempty"`
+	Timeout               time.Duration `mapstructure:"timeout,omitempty"`
+}
+
+// KeepaliveEnforcementPolicy allow configuration of the keepalive.EnforcementPolicy.
+// The same default values as keepalive.EnforcementPolicy are applicable and get applied by the server.
+// See https://godoc.org/google.golang.org/grpc/keepalive#EnforcementPolicy for details.
+type KeepaliveEnforcementPolicy struct {
+	MinTime             time.Duration `mapstructure:"min_time,omitempty"`
 	PermitWithoutStream bool          `mapstructure:"permit_without_stream,omitempty"`
 }
 
-// GrpcSettingsToDialOptions maps configgrpc.GRPCClientSettings to a slice of dial options for gRPC
-func GrpcSettingsToDialOptions(settings GRPCClientSettings) ([]grpc.DialOption, error) {
+type GRPCServerSettings struct {
+	// Configures the generic server protocol.
+	configprotocol.ProtocolServerSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
```

Comment:
Is it worth putting consideration into how this will look in cases where we support multiple protocols over the same endpoint? e.g. gRPC Gateway or the Jaeger receiver config (maybe gRPC Gateway is a special case where we can just continue to specificy `CorsOrigins` separate to the gRPC settings)

Reply:
Also it seems weird to me that in gRPC Client Settings, TLS settings are squashed, but in gRPC Server settings, TLS settings are under `tls_configuration` (inside `ProtocolServerSettings`). Is there any possibilty we can make that more consistent?

Reply:
Will fix that in a future PR.

Reply:
Alright - I assume you're referring to the second comment.

Any thoughts on the first comment?

Reply:
Both will come soon :)